web/src/components/Upload/Upload/Upload.tsx

```diff
@@ -55,8 +77,44 @@ const Upload = ({ upload, queryResult }) => {
     downloadUploadFile({ variables: { id: upload.id } })
   }
 
+  const [createValidation] = useMutation(CREATE_VALIDATION_MUTATION, {
+    onCompleted: async () => {
+      toast.success('Upload invalidated')
+      await refetch()
+      setSavingUpload(false)
+    },
+    onError: (error) => {
+      toast.error(error.message)
+      setSavingUpload(false)
+    },
+  })
+
   const handleValidate = () => {}
-  const handleForceInvalidate = () => {}
+
+  const handleForceInvalidate = async () => {
+    setSavingUpload(true)
+    const invalidateResult = [
+      {
+        message: `Manually invalidated by User: ${currentUser.name}`,
```

Comment:
can we also add `currentUser.email` here? Just in the event that two users have the same name.

Reply:
yes! added